pubsub/pubsub.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pubsub
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/google/go-cloud/pubsub/driver"
+	"google.golang.org/api/support/bundler"
+)
+
+// Message contains data to be published.
+type Message struct {
+	// Body contains the content of the message.
+	Body []byte
+
+	// Attributes has key/value metadata for the message.
+	Attributes map[string]string
+
+	// AckID identifies the message on the server.
+	// It can be used to ack the message after it has been received.
+	ackID AckID
+
+	// sub is the Subscription this message was received from.
+	sub *Subscription
+}
+
+// AckID identifies a message for acknowledgement.
+type AckID interface{}
+
+// Ack acknowledges the message, telling the server that it does not need to
+// be sent again to the associated Subscription. This method blocks until
+// the message has been confirmed as acknowledged on the server, or failure
+// occurs.
```

Comment:
Add the same comment as you have on `Send` about it being able to be called from multiple goroutines at once.